src/server/scripts/Northrend/Nexus/EyeOfEternity/boss_malygos.cpp

```diff
@@ -1201,14 +1201,18 @@ class npc_scion_of_eternity : public CreatureScript
                             for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
                                 if (Player* pPlayer = i->GetSource())
                                 {
-                                    if (!pPlayer->IsAlive() || pPlayer->IsGameMaster())
-                                        continue;
-                                    guids.push_back(pPlayer->GetGUID());
+                                    if (pPlayer->IsAlive())
+                                    {
+                                        if (!pPlayer->GetVehicle())
+                                        {
+                                            guids.push_back(pPlayer->GetGUID());
+                                        }
+                                    }
                                 }
                         if (!guids.empty())
                             if (Player* plr = ObjectAccessor::GetPlayer(*me, guids.at(urand(0, guids.size() - 1))))
                                 me->CastSpell(plr, SPELL_SCION_ARCANE_BARRAGE);
-
+                        guids.clear();
```

Comment:
`guids` is a local variable, it only exists inside this scope, pointless to clear